apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java

```diff
@@ -81,116 +81,25 @@ private Optional<RetentionConfig> getTableRetention(GetTableResponseBody respons
-    GetTableResponseBody response = getTable(tableMetadata);
-    return response != null && checkCreationTimeEligibility(response) && isPrimaryTable(response);
-  }
-
-  /**
-   * Checks if expire snapshots can be executed on the input table.
-   *
-   * @param tableMetadata table metadata
-   * @return true if the table can expire snapshots, false otherwise
-   */
-  public boolean canExpireSnapshots(TableMetadata tableMetadata) {
-    GetTableResponseBody response = getTable(tableMetadata);
-    return response != null && checkCreationTimeEligibility(response) && isPrimaryTable(response);
-  }
-
-  /**
-   * Checks if retention can be executed on the input table.
-   *
-   * @param tableMetadata table metadata
-   * @return true if the table can run retention, false otherwise
-   */
-  public boolean canRunRetention(TableMetadata tableMetadata) {
-    GetTableResponseBody response = getTable(tableMetadata);
-
-    if (response == null || !checkCreationTimeEligibility(response) || !isPrimaryTable(response)) {
-      return false;
-    }
-    Optional<RetentionConfig> config = getTableRetention(response);
-    return config.isPresent();
-  }
-
-  /**
-   * Checks if stats collection can be executed on the input table.
-   *
-   * @param tableMetadata table metadata
-   * @return true if the stats collection can happen, false otherwise
-   */
-  public boolean canRunTableStatsCollection(TableMetadata tableMetadata) {
-    GetTableResponseBody response = getTable(tableMetadata);
-    return response != null && checkCreationTimeEligibility(response);
-  }
-
-  /**
-   * Checks if staged deletion task can be run on given table
-   *
-   * @param tableMetadata
-   * @return
-   */
-  public boolean canRunStagedDataDeletion(@NonNull TableMetadata tableMetadata) {
-    GetTableResponseBody response = getTable(tableMetadata);
-    return response != null && checkCreationTimeEligibility(response);
-  }
-
-  /**
-   * Checks if orphan files deletion task can be run on given table
-   *
-   * @param tableMetadata
-   * @return
-   */
-  public boolean canRunOrphanFileDeletion(@NonNull TableMetadata tableMetadata) {
-    GetTableResponseBody response = getTable(tableMetadata);
-    return response != null && checkCreationTimeEligibility(response);
-  }
-
-  private boolean checkCreationTimeEligibility(@NonNull GetTableResponseBody response) {
-    if (response.getCreationTime() == null) {
-      // no creationTime assigned to table, by default we pick these tables for maintenance
-      return true;
-    }
-    return response.getCreationTime()
-        < System.currentTimeMillis() - TimeUnit.HOURS.toMillis(cutOffHours);
-  }
-
-  private boolean isPrimaryTable(@NonNull GetTableResponseBody response) {
-    return GetTableResponseBody.TableTypeEnum.PRIMARY_TABLE == response.getTableType();
-  }
-
-  public List<TableMetadata> getTables() {
-    List<TableMetadata> ret = new ArrayList<>();
+  public List<TableMetadata> getTableMetadataList() {
```

Comment:
Add documentation for all new functions?

Reply:
The above one is very simple, this is much more complex, retrieving first databases and then all tables within that database. I'd just add a simple 'Iteratively retrieves the metadata for all registered tables and transforms them to objects of the TableMetadata type.' or something along the lines. It's just for readability so that as a someone trying to understand the code, I don't have to parse 30 LoC with special function calls.

Reply:
Added.